src/table.js

```diff
@@ -538,15 +538,88 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
```

Comment:
In the case of numbers, we probably want the default value to be NaN (or undefined) instead of null. If you expect everything to be numbers by asserting that the type is number, and we sometimes provide nulls, then the nulls are likely to be silently coerced to zeroes. It’s usually better to coerce them to NaN so they are properly numbers, and/or undefined which implicitly gets coerced to NaN when you perform arithmetic, so that missing values are silently replaced with zeroes and potentially corrupting calculations.

For the same reason, we might want dates to be represented as undefined or new Date(NaN) (“invalid date”)…

Reply:
Thanks! Revised as shown below. 
Soft-coerce|Hard-coerce
-|-
![Screen Shot 2023-01-26 at 10 33 00 AM](https://user-images.githubusercontent.com/111310561/214964468-f05ca7f3-de21-4dce-91cb-394746e50c5c.png)|![Screen Shot 2023-01-26 at 10 33 13 AM](https://user-images.githubusercontent.com/111310561/214964503-f343e3a6-b96f-49a9-8915-71e1b425e974.png)
![Screen Shot 2023-01-26 at 11 07 19 AM](https://user-images.githubusercontent.com/111310561/214964535-40bd2056-65bf-4631-a6aa-9e2414b9c529.png)|![Screen Shot 2023-01-26 at 11 07 57 AM](https://user-images.githubusercontent.com/111310561/214964554-563c894a-5329-439e-a196-879c4cd818df.png)